src/app/ReadClient.cpp

```diff
@@ -253,7 +266,51 @@ CHIP_ERROR ReadClient::SendReadRequest(ReadPrepareParams & aReadPrepareParams)
             }
         }
 
-        ReturnErrorOnFailure(request.IsFabricFiltered(aReadPrepareParams.mIsFabricFiltered).EndOfReadRequestMessage().GetError());
+        ReturnErrorOnFailure(request.IsFabricFiltered(aReadPrepareParams.mIsFabricFiltered).GetError());
+
+        bool hasEncodeDataVersionList = false;
```

Comment:
Is this actually `encodedDataVersionFilters`?  Or more to the point `unreservedTLVEncodingOverhead`?

Reply:
the latter for unreservedTLVEncodingOverhead, just largely simplified this part, thanks